src/sys/socket/mod.rs

```diff
@@ -1796,90 +1792,10 @@ pub fn getsockname(fd: RawFd) -> Result<SockAddr> {
 
         Errno::result(ret)?;
 
-        sockaddr_storage_to_addr(&addr.assume_init(), len as usize)
+        SockAddr::from_raw_sockaddr(addr.as_ptr() as _, len as usize)
     }
 }
 
-/// Return the appropriate `SockAddr` type from a `sockaddr_storage` of a
-/// certain size.
-///
-/// In C this would usually be done by casting.  The `len` argument
-/// should be the number of bytes in the `sockaddr_storage` that are actually
-/// allocated and valid.  It must be at least as large as all the useful parts
-/// of the structure.  Note that in the case of a `sockaddr_un`, `len` need not
-/// include the terminating null.
-pub fn sockaddr_storage_to_addr(
```

Comment:
Ok, I get the motivation for removing this now.  It's not that it's inherently `unsafe`, it's that it's difficult to use safely, because they signature requires a fully initialized `sockaddr_storage` but the caller often won't have that.  Alternatives would be to use a `*const sockaddr_storage` or a `mem::MaybeUninit<&sockaddr_storage>`, both of which would have to be coupled with a `len`.  The latter is probably better, because it at least checks the argument's lifetime at compile-time.
Two things you definitely should do are:
* Explain this in the commit message
* `#[deprecate]` the old function rather than remove it.